### PR TITLE
Update warp_affine test and docs

### PR DESCRIPTION
**Pull request by JanuszL**
- change relative error in warp affine test to the max_error
- changes `fill_value` default value to empty one
- adds a note that [0, 0] coordinate in the warp affine
  transformation equation means the corner of the first pixel

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Description
- [ ] **Bug fix** (*non-breaking change which fixes an issue*)
- [ ] **New feature** (*non-breaking change which adds functionality*)
- [ ] **Breaking change** (*fix or feature that would cause existing functionality to not work as expected*)
- [ ] **Refactoring** (*Redesign of existing code that doesn't affect functionality*)
- [x] **Other** (*e.g. Documentation, Tests, Configuration*)

### What happened in this PR
- change relative error in warp affine test to the max_error
- changes `fill_value` default value to empty one
- adds a note that [0, 0] coordinate in the warp affine
  transformation equation means the corner of the first pixel

#### Additional information
- Affected modules and functionalities:
  - warp_affine schema
  - warp_attr schema
  - warp_affine tests

- Key points relevant for the review:
  - NA

### Tests
- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [x] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

relates to https://github.com/NVIDIA/DALI/issues/3527

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

File: dali/operators/image/remap/warp_affine.cc [hunk 1]
```diff
@@ -17,25 +17,30 @@
...
 With a list of values ``(M11, M12, M13, M21, M22, M23)``, this operation produces a new image
 by using the following formula::
 
   dst(x,y) = src(M11 * x + M12 * y + M13, M21 * x + M22 * y + M23)
 
-If the ``inverse_map`` option is set to false, the matrix represents a source to destination 
+Where ``[0, 0]`` coordinate means the corner of the first pixel.
+
+If the ``inverse_map`` option is set to false, the matrix represents a source to destination
 transform and it is inverted before applying to the formula above.
```
Comment (mzient) [inline dali/operators/image/remap/warp_affine.cc:39]:
```suggestion
transform and it is inverted before applying the formula above.
```

File: dali/operators/image/remap/warp_attr.cc [hunk 2]
```diff
@@ -29,7 +29,7 @@ be excluded (for example, for RGB images, specify ``(480,640)``, not ``(480,640,
   .AddOptionalArg("fill_value", R"code(Value used to fill areas that are outside the source image.
 
 If a value is not specified, the source coordinates are clamped and the border pixel is
-repeated.)code", 0.0f)
+repeated.)code", DALI_FLOAT)
```
Comment (mzient) [inline dali/operators/image/remap/warp_attr.cc:32]: :+1:

File: dali/operators/image/remap/warp_affine.cc (hunk 1 above)
Comment (JanuszL) [reply dali/operators/image/remap/warp_affine.cc]: Done

**Comment by JanuszL** [comment]
!build

**Review by mzient — APPROVED** [review]

**Comment by JanuszL** [comment]
!build

**Review by jantonguirao — APPROVED** [review]

**Pull request merged**